lib/dynflow/world.rb

```diff
@@ -32,7 +33,7 @@ def default_options
     end
 
     def clock
-      @clock ||= Clock.new(logger)
+      @clock_barrier.synchronize { @clock ||= Clock.new(logger) }
```

Comment:
I was thinking about it, but in the remote executor case, the clock is seldom used: so we will have additional thread in every client instance.